vortex-array/src/compress.rs

```diff
@@ -1,221 +1,30 @@
-use std::collections::HashSet;
-use std::fmt::{Debug, Display, Formatter};
+use vortex_error::VortexResult;
 
-use log::{debug, info, warn};
-use vortex_error::{vortex_bail, VortexResult};
+use crate::Array;
 
-use crate::array::chunked::{Chunked, ChunkedArray};
-use crate::array::constant::{Constant, ConstantArray};
-use crate::array::struct_::{Struct, StructArray};
-use crate::compute::slice::slice;
-use crate::compute::unary::scalar_at::scalar_at;
-use crate::encoding::{ArrayEncoding, EncodingRef};
-use crate::sampling::stratified_slices;
-use crate::stats::ArrayStatistics;
-use crate::validity::Validity;
-use crate::{Array, ArrayDType, ArrayDef, ArrayTrait, Context, IntoArray, IntoCanonical};
-
-pub trait EncodingCompression: ArrayEncoding {
-    fn cost(&self) -> u8 {
-        1
-    }
-
-    fn can_compress(
-        &self,
-        _array: &Array,
-        _config: &CompressConfig,
-    ) -> Option<&dyn EncodingCompression> {
-        None
-    }
-
-    fn compress(
-        &self,
-        _array: &Array,
-        _like: Option<&Array>,
-        _ctx: Compressor,
-    ) -> VortexResult<Array> {
-        vortex_bail!(NotImplemented: "compress", self.id())
-    }
-
-    // For an array returned by this encoding, give the size in bytes minus any constant overheads.
-    fn compressed_nbytes(&self, array: &Array) -> usize {
-        array.with_dyn(|a| a.nbytes())
-    }
-}
-
-#[derive(Debug, Clone)]
-pub struct CompressConfig {
-    #[allow(dead_code)]
-    block_size: u32,
-    sample_size: u16,
-    sample_count: u16,
-    max_depth: u8,
-    // TODO(ngates): can each encoding define their own configs?
-    pub ree_average_run_threshold: f32,
+pub trait CompressionStrategy {
+    fn compress(&self, array: &Array) -> VortexResult<Array>;
 }
 
-impl Default for CompressConfig {
-    fn default() -> Self {
-        // TODO(ngates): we should ensure that sample_size * sample_count <= block_size
-        Self {
-            block_size: 65_536,
-            // Sample length should always be multiple of 1024
-            sample_size: 128,
-            sample_count: 8,
-            max_depth: 3,
-            ree_average_run_threshold: 2.0,
-        }
-    }
-}
-
-#[derive(Debug, Clone)]
 pub struct Compressor<'a> {
```

Comment:
Does Compressor still serve a purpose? It seems like it's mostly supplanted by `CompressionStrategy`

Reply:
Not sure really, I thought there's some random utility to share but seems like the answer is mostly no